include/oneapi/dpl/pstl/utils.h

```diff
@@ -128,6 +128,20 @@ class __pstl_assign
     }
 };
 
+template <typename _Comp, typename _Proj>
+struct __compare
```

Comment:
I wonder if this should be named `__project_and_compare` or `__compare_with_projection`, or `__projected_compare`
`__compare` seems too general, and doesn't actually describe the purpose behind this structure.  The reason one would use this is to apply the projection as input to a comparison.

Reply:
Actually, I thought about that... about the similar names that you mention...
But came to conclusion:
1) any of them is too long.
2) consistency reasons:  we have many  other functor with additional fields like predicate or/and value and we don't  mention that fields in the functor name. We dont write  `__replace_functor` `_with_predicate_and_value`...
3) as for me I think even `__replace_functor `or `__transform_functor `or `__pstl_less` is too long and "functor" (or pstl) word doesn't add any sense:) it's the same if we write `transform_struct `or `transform_class` or `sort_function` :) The fact that `__transform_functor`  is a functor so is clear from context call transform{pred}() or class definition. So, in my opinion `__transform`, `__compare` etc is absolutely enough. Moreover we have ones in the special internal namespace, if you scare about a name conflict.
4) The other libraries also don't  use a kind of additional suffix word for the predicates: sycl::plus, sycl::bit_or etc.

Reply:
I'm not worried about a name conflict.  I just think it is better for understanding the code later, but its just my preference.  Especially with this being internal, I prefer more descriptive names to concise ones. 

I wont insist, not a blocker.